Sources/MapboxMaps/Location/Puck/Puck2D.swift

```diff
@@ -57,13 +62,15 @@ internal final class Puck2D: NSObject, Puck {
             stretchX: [],
             stretchY: [],
             content: nil)
-        try! style.addImage(
-            configuration.resolvedBearingImage,
-            id: Self.bearingImageId,
-            sdf: false,
-            stretchX: [],
-            stretchY: [],
-            content: nil)
+        if showBearingImage {
+            try! style.addImage(
+                configuration.resolvedBearingImage,
+                id: Self.bearingImageId,
+                sdf: false,
+                stretchX: [],
+                stretchY: [],
+                content: nil)
+        }
```

Comment:
In addition to conditionally adding the image, I think you also need to conditionally set bearingImage on the layer near line 93 below

Reply:
Do you mean remove the condition check here, only check it in line 93?

Reply:
I think we need both

Reply:
Yeah, that's right.